src/AzureClient/AzureClient.cs

```diff
@@ -39,7 +43,12 @@ public class AzureClient : IAzureClient
         private AzureExecutionTarget? ActiveTarget { get; set; }
         private string MostRecentJobId { get; set; } = string.Empty;
         private IEnumerable<ProviderStatusInfo>? AvailableProviders { get; set; }
-        private IEnumerable<TargetStatusInfo>? AvailableTargets => AvailableProviders?.SelectMany(provider => provider.Targets);
+        private IEnumerable<TargetStatusInfo>? AvailableTargets => 
+            AvailableProviders
+            ?.SelectMany(provider => provider.Targets)
+            ?.Where(t => t.TargetId != null
+                && (t.TargetId.StartsWith(MicrosoftSimulator) // Of the Microsoft targets, we only want those that start with 'microsoft.simulator'.
+                || AzureExecutionTarget.GetProvider(t.TargetId) != AzureProvider.Microsoft)); // We want all non-Microsoft targets.
```

Comment:
```suggestion
            ?.Where(t => t.TargetId != null ? IsQuantumExecutionTarget(t.TargetId) : false);
```

Reply:
This suggestion uses the method mentioned in [this comment](https://github.com/microsoft/iqsharp/pull/601/files#r831420468).

Reply:
I've added something similar to this suggestion, but I think it is simpler without the ternary operators and to just use regular Boolean operators.

Reply:
The `MicrosoftSimulator` string will become even more important in the upcoming PR for the output formatting.